macros/dedupe_artifacts_v1.sql

```diff
@@ -0,0 +1,29 @@
+{% macro dedupe_dbt_artifacts_v1() %}
+
+{% set src_dbt_artifacts = source('dbt_artifacts', 'artifacts') %}
+
+{% set dedupe_results_query %}
+
+    create temporary table {{ src_dbt_artifacts.database }}.{{ src_dbt_artifacts.schema }}.dbt_temp_artifact_table as
+        select * from {{ src_dbt_artifacts }}
+        qualify row_number() over (
+            partition by generated_at
+            -- NB: Snowflake requires an order by clause, although all rows will be the same within a partition.
+            order by generated_at
+        ) = 1;
+    
+    truncate {{ src_dbt_artifacts }};
+
+    insert into {{ src_dbt_artifacts }} select * from {{ src_dbt_artifacts.database }}.{{ src_dbt_artifacts.schema }}.dbt_temp_artifact_table;
+
+    drop table {{ src_dbt_artifacts.database }}.{{ src_dbt_artifacts.schema }}.dbt_temp_artifact_table;

```

Comment:
```suggestion
    create or replace table {{ src_dbt_artifacts }} clone {{ artifact_table.database }}.{{ artifact_table.schema }}.dbt_temp_artifact_table;
```

Reply:
Good idea 👍  (although in this case we still need the `drop`). I think I've implemented what you _meant_ rather than this exact suggestion.